### PR TITLE
LR assignement tutorial

### PR DESCRIPTION
**Pull request by shaydeci**
Added .md tutorial for lr assignment and finetune hyperparameter.

### CODE REVIEW

File: documentation/source/LRAssignment.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Assigning Learning Rates in SG
+The `initial_lr` training hyperparameter allows you to specify different learning rates for different layers or groups of parameters in your neural network. This can be particularly useful for fine-tuning pre-trained models or when different parts of your model require different learning rate settings for optimal training.
+
+## Using initial_lr as a Scalar:
+When initial_lr is a single floating-point number, it sets a uniform learning rate for all model parameters. For example, initial_lr = 0.01:
```
Comment (BloodAxe) [inline documentation/source/LRAssignment.md:5]: When referring to a variable name from the text, it's cool to use ` quotes to highlight the variable itself: `initial_lr`, `initial_lr = 0.01`

File: documentation/source/LRAssignment.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+# Train the model
+trainer.train(model, training_params, train_dataloader, test_dataloader)
+```
+
+
+
+
+### How `finetune` Works
+
+- When `finetune` is set to `True`, the model automatically freezes a part of itself based on the definitions in the `get_finetune_lr_dict` method.
```
Comment (BloodAxe) [inline documentation/source/LRAssignment.md:114]: Maybe at the end would be helpful to provide a code snippet showing how this method implement for YoloNAS, just to show the code to complement the text description.

**Review by BloodAxe — APPROVED** [review]
LGTM

**Review by BloodAxe — APPROVED** [review]

File: documentation/source/LRAssignment.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+# Define model
+model = 
+
+# Define data loaders
+train_dataloader = ...
+test_dataloader = ...
+
+# Define training parameters
+training_params = {
+    "initial_lr": 0.01,
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/LRAssignment.md]: Updated https://github.com/Deci-AI/super-gradients/commit/b244ec0db1f8427650c3918bf1987cb4f7833f30

File: documentation/source/LRAssignment.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+for those specific groups. This approach offers granular control over the learning rates for different parts of the model.
+
+* Each key in the initial_lr dictionary acts as a prefix to match the named parameters in the model. The learning rate associated with a key is applied to all parameters whose names start with that prefix.
+
+* The "default" key is essential, as it provides a fallback learning rate for any parameter that does not match other specified prefixes.
+  
+* Freezing parameters can be done by assigning a learning rate of 0 to a specific prefix. By doing so, you will be preventing them from being updated during training.
+
+
+For example, in the below snippet conv1 and conv2 will be frozen, and fc1 and fc2 will be trained with an initial learning rate of 0.001:
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/LRAssignment.md]: Updated https://github.com/Deci-AI/super-gradients/commit/b244ec0db1f8427650c3918bf1987cb4f7833f30

File: documentation/source/LRAssignment.md (hunk 2 above)
Comment (shaydeci) [reply documentation/source/LRAssignment.md:114]: Updated https://github.com/Deci-AI/super-gradients/commit/b244ec0db1f8427650c3918bf1987cb4f7833f30
